other/alarms.php

```diff
@@ -395,39 +399,33 @@ function insert_new(){
     "message, " .
     "recipients_json, " .
     "locked, " .
+    "visible, " .
     "description, " .
     "subject, " .
     "active" .
-    ") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
-  $statement = $dbi->prepare($query);
-  # bind parameters for markers, where (s = string, i = integer, d = double,
-  #                                     b = blob)
-
-  $locked = 0;
-  $statement->bind_param('sssississi',
-			 $data["quiries_json"],
-			 $data["parameters_json"],
-			 $data["check"],
-			 $data["no_repeat_interval"],
-			 $data["message"],
-			 $data["recipients_json"],
-			 $locked,
-			 $data["description"],
-			 $data["subject"],
-			 $data["active"]
-			 );
+    ") VALUES (?, ?, ?, ?, ?, ?, 0, 1, ?, ?, ?)";
+  // New rules are never locked and always visible
+  $statement = $db->prepare($query);
+
+  $statement->bindParam(1, $data["quiries_json"], PDO::PARAM_STR);
```

Comment:
Does this have a version where you bind via the name. Would make it **a lot** easier to see what is going on.

Reply:
Like, I almost made a comment about the numbers being of from "description" and forwards until I noticed that "locked" and "visible" are hard coded.

Reply:
since it is a prepared statement, I believe it can be done by name: https://www.php.net/manual/en/pdostatement.bindparam.php
This, however, is a general cleanup that can be done in a later PR going through the other scripts as well. As it stands, the new syntax is more readable than the old even keeping it bound by index

Reply:
although I am not entirely sure with the ? placeholders..

Reply:
🏳️ ok by me, just raised it as a readability concern